src/Sushi.Mediakiwi.API/Services/ContentService.cs

```diff
@@ -1428,6 +1605,39 @@ public async Task<GetListResponse> GetListResponseAsync(UrlResolver resolver)
                         result.FormMaps = new List<Transport.FormMap>();
                     }
                     result.FormMaps.AddRange(formMaps);
+
+                    // Loop through any Datalists
+                    foreach (var formMap in formMaps.Where(x => x.Fields?.Any(x => x.ContentType == ContentTypeEnum.DataList) == true))
+                    {
+                        foreach (var dataList in formMap.Fields.Where(x => x.ContentType == ContentTypeEnum.DataList))
```

Comment:
Please move the Where clauses outside the foreach loops to reduce complexity from quadratic to linear